### PR TITLE
add graphics tutorial

### PR DESCRIPTION
**Pull request by techiepriyansh**
Add graphics tutorial in docs

### CODE REVIEW

File: docs/guides/graphics_tutorial.rst [hunk 1]
```diff
@@ -0,0 +1,236 @@
+=================
+Graphics Tutorial
```
Comment (Ishan-002) [inline docs/guides/graphics_tutorial.rst]: Rename this to something like ```Exploring the Graphics capabilities of Rootex```

**Review by Ishan-002 — CHANGES_REQUESTED** [review]

**Review by Ishan-002 — CHANGES_REQUESTED** [review]
The tutorial looks pretty good with self-explanatory images! The only problem is that the instructions feel like a set of commands and may be a little hard to follow. Instead, write the tutorial in flow-like manner, where you're explaining things and the procedure along with doing stuff on the editor GUI. Look at #560 for reference.

**Comment by Ishan-002** [comment]
The tutorial is written in such a style that all the instructions written at one place and then 4-5 images are shown at one go which tell about the process. Instead, change this to 1-2 instructions and then the subsequent image for it.

File: docs/guides/graphics_tutorial.rst [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+	:align: center
+
+
+***************
+Decal Component
+***************
+
+To add a decal component. 
+
+1) Make a component DECAL and give it transform and Decal Component.
```
Comment (Ishan-002) [inline docs/guides/graphics_tutorial.rst]: Make a ~~component~~ scene DECAL

File: docs/guides/graphics_tutorial.rst [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+2) Click Edit Components.
+3) Check the appropriate components, in this case, transform and Model. **Note**: Transform Component is a must.
+
+.. image:: images/graphics_tutorial/add_model_component_and_transform_component.png
+	:alt: Components Added
+	:align: center
+	
+4) Open inspector
+5) Go to the model component in the inspector.
+6) Click the folder icon next to Model.
```
Comment (Ishan-002) [inline docs/guides/graphics_tutorial.rst:63]: Add an instruction telling the path of the sponza file to go to.

File: docs/guides/graphics_tutorial.rst [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+
+For sponza initially, it would look like this:
+
+.. image:: images/graphics_tutorial/model_file_just_added.png
+	:alt: Model File Just Added
+	:align: center
+
+|
+
+This is due to the default settings of the sponza obj file. To get a better view, set the scale to (0.0031, 0.031, 0.031) and set the LOD distance to 123:
```
Comment (Ishan-002) [inline docs/guides/graphics_tutorial.rst]: 0.0031 -> 0.031 for x scaling

File: docs/guides/graphics_tutorial.rst [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+
+1) Create a new material by going to file -> Create Resource.
+2) Name the material and click create.
+
+.. image:: images/graphics_tutorial/create_basic_material.png
+	:alt: Create Basic Material
+	:align: center
+
+3) Go to the `inspector-> model component->Materials`.
+4) click on the folder icon on the corresponding overriding material.
```
Comment (Ishan-002) [inline docs/guides/graphics_tutorial.rst]: Again, tell the filepath of the material

File: docs/guides/graphics_tutorial.rst [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+
+
+*********************
+Overriding a material
+*********************
+
+To change the properties of one object without changing the original material, we can use overriding materials. To override a material:
+
+1) Create a new material by going to file -> Create Resource.
+2) Name the material and click create.
```
Comment (Ishan-002) [inline docs/guides/graphics_tutorial.rst:178]: Tell the type of material too.

File: docs/guides/graphics_tutorial.rst [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+
+.. image:: images/graphics_tutorial/change_custom_material_pixel_shader.png
+	:alt: Change Custom Material Pixel Shader
+	:align: center
+
+You can use fire_pixel_shader from rootex/core/renderer/shaders
+
+Clicking on the pencil icon opens an editor to customise the shader.
+
+**Note:** You can only add shaders to custom materials. If you want to use default material, override the original default material with custom material and then add a shader to the overriding material. The overriding material does inherit the textures of the original materials.
```
Comment (Ishan-002) [inline docs/guides/graphics_tutorial.rst]: You can use use a box for writing notes.
Refer: https://github.com/sdslabs/rootex/blob/dev/docs/guides/getting_started.rst

File: docs/guides/graphics_tutorial.rst [hunk 8]
```diff
@@ -0,0 +1,279 @@
...
+	:alt: Add Decal Component
+	:align: center
+2) Create a decal material. By going to File -> CreateResource. And then slect Decal material in resource type dropdown.
+
+.. image:: images/graphics_tutorial/create_decal_material.png
+	:alt: Create Decal Material
+	:align: center
+	
+2) Now go to the inspector and click DecalComponent. 
+3) Click on the folder icon and select the decal material.
```
Comment (Ishan-002) [inline docs/guides/graphics_tutorial.rst]: Numbering is wrong

**Review by Ishan-002 — CHANGES_REQUESTED** [review]
Add a blank line after the last image
image

File: docs/guides/graphics_tutorial.rst (hunk 3 above)
Comment (techiepriyansh) [reply docs/guides/graphics_tutorial.rst:63]: It is visible in the screenshot

File: docs/guides/graphics_tutorial.rst (hunk 5 above)
Comment (techiepriyansh) [reply docs/guides/graphics_tutorial.rst]: Again, it's visible in the screenshot

File: docs/guides/graphics_tutorial.rst (hunk 3 above)
Comment (Ishan-002) [reply docs/guides/graphics_tutorial.rst:63]: It's better that it is written in the text too. Else, you'd have to zoom in to look at the filepath.

File: docs/guides/graphics_tutorial.rst (hunk 5 above)
Comment (Ishan-002) [reply docs/guides/graphics_tutorial.rst]: Better to have it in text too. Makes the tutorial reading easy.

Comment (techiepriyansh) [reply docs/guides/graphics_tutorial.rst]: done

**Review by Ishan-002 — CHANGES_REQUESTED** [review]
You can add spaces, in between images and text at some places where they look too close. 
See this: https://github.com/sdslabs/Rootex/pull/561#issuecomment-1052380775

**Review by Ishan-002 — APPROVED** [review]
